test/topology_custom/test_topology_failure_recovery.py

```diff
@@ -35,13 +40,12 @@ async def test_tablet_drain_failure_during_decommission(manager: ManagerClient):
     keys = range(256)
     await asyncio.gather(*[cql.run_async(f"INSERT INTO test.test (pk, c) VALUES ({k}, {k});") for k in keys])
 
-    await inject_error_on(manager, "stream_tablet_fail_on_drain", servers)
+    async for e in inject_error_on(manager, "stream_tablet_fail_on_drain", servers):
```

Comment:
Is this use of "async for" a common idiom?  the need to specify a name "e" which you don't use is especially misleading. Wouldn't a contextmanager and "with" be a more familiar idiom?

Reply:
I'm not sure how to implement an async context manager.
The `async for` idiom is apparently the common practice for async generators (that yield values) in python

Reply:
> the need to specify a name "e" which you don't use is especially misleading

I can rename it to `_` if it makes it clearer that it's unused (but still needs a scope to hold the yielded value before it's destroyed)

Reply:
in v2: extended pylib/rest_client by implementing `inject_error_on` using a `@asynccontextmanager`